activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -608,10 +608,10 @@ def primary_keys(table_name) # :nodoc:
       end
 
       def case_sensitive_comparison(table, attribute, column, value)
-        if value.nil? || column.case_sensitive?
-          super
-        else
+        if !value.nil? && column.collation && !column.case_sensitive?
```

Comment:
Doesn't this mean `case_sensitive?` is lying?

Looks like [this](https://github.com/rails/rails/blob/127fd4705b02e4f2d6d1bf544f7c116de8adcdd9/activerecord/lib/active_record/connection_adapters/mysql/column.rb#L27) should be `!collation || ..`